controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -451,38 +571,130 @@ func (pp *portPublisher) updateEndpoints(endpoints *corev1.Endpoints) {
 			}
 		}
 	}
-	pp.exists = true
 	pp.addresses = newAddressSet
+}
 
-	pp.metrics.incUpdates()
-	pp.metrics.setPods(len(pp.addresses.Addresses))
-	pp.metrics.setExists(true)
+func (pp *portPublisher) updateEndpointSlicesSet(endpointslice *dv1beta1.EndpointSlice) {
+	newAddressSet := pp.endpointSliceToAddresses(endpointslice)
+	if len(newAddressSet.Addresses) == 0 {
+		for _, listener := range pp.listeners {
+			listener.NoEndpoints(true)
```

Comment:
I'm not sure the logic is correct here for dealing with individual EndpointSlice objects.  If a single EndpointSlice object has no addresses, I think there still could be other EndpointSlice objects for that service that do have addresses.  So I don't think a NoEndpoints update is appropriate.

Reply:
that is totally correct, I think I might have confused myself at the time. Thank you for spotting this, I have replaced the above with a guard clause to return out of the function in case the new address set has a length of zero. Would this need any other special consideration? wdyt